File: java code try/TCS Coding/VersionCompare.c
#include <stdio.h>

int getNumber(char *str, int *i) {
    int num = 0;

    while (str[*i] && str[*i] != '.') {
        num = num * 10 + (str[*i] - '0');
        (*i)++;
    }

    // Skip dot ONLY if present
    if (str[*i] == '.')
        (*i)++;

    return num;
}

void compareVersion(char *v1, char *v2) {
    int i = 0, j = 0;

    while (v1[i] || v2[j]) {

        int num1 = v1[i] ? getNumber(v1, &i) : 0;
        int num2 = v2[j] ? getNumber(v2, &j) : 0;

        if (num1 < num2) {
            printf("Upgraded\n");
            return;
        }

        if (num1 > num2) {
            printf("Downgraded\n");
            return;
        }
    }

    printf("Same Version\n");
}

int main() {
    compareVersion("4.8.2", "4.8.4");
    compareVersion("4.0.2", "4.8.4");
    compareVersion("4.8.2", "4.8.2.1");
    return 0;
}

File: java code try/TCS Coding/VersionCompare.java
public class VersionCompare {

    public static String compareVersion(String v1, String v2) {

        String[] a = v1.split("\\.");
        String[] b = v2.split("\\.");

        int n = Math.max(a.length, b.length);

        for (int i = 0; i < n; i++) {
            int num1 = i < a.length ? Integer.parseInt(a[i]) : 0;
            int num2 = i < b.length ? Integer.parseInt(b[i]) : 0;

            if (num1 < num2) return "Upgraded";
            if (num1 > num2) return "Downgraded";
        }

        return "Same Version";
    }

    public static void main(String[] args) {
        System.out.println(compareVersion("4.8", "4.8.4"));
        System.out.println(compareVersion("4.0.2", "4.8.4"));
        System.out.println(compareVersion("4.8.2", "4.8.2"));
    }
}